src/linkPanel/widgets/linking.tsx

```diff
@@ -0,0 +1,745 @@
+// import { ToolbarRegistry } from '@jupyterlab/apputils';
+// import { ObservableList } from '@jupyterlab/observables';
+import {
+  Button,
+  ReactWidget,
+  UseSignal
+  // Toolbar,
+  // ToolbarButton
+} from '@jupyterlab/ui-components';
+import { Widget } from '@lumino/widgets';
+import * as React from 'react';
+
+import { IGlueSessionSharedModel } from '../../types';
+import { LinkEditorWidget } from '../linkEditorWidget';
+import {
+  ComponentLinkType,
+  // IAdvLinkCategories,
+  // IAdvLinkDescription,
+  ILink,
+  ILinkEditorModel,
+  IDatasets,
+  IDatasetsKeys,
+  // IdentityLinkFunction,
+  IdentityLinkUsing
+} from '../types';
+import { ISignal, Signal } from '@lumino/signaling';
+
+export class Linking extends LinkEditorWidget {
+  constructor(options: LinkEditorWidget.IOptions) {
+    super(options);
+
+    this.addClass('glue-LinkEditor-linking');
+
+    // this._selectedAdvLink = { category: '', linkName: '' };
+    this._datasetsPanels = [
+      this._emptyDatasetPanel(),
+      this._emptyDatasetPanel()
+    ];
+
+    this._attributesPanels = [
+      this._emptyAttributePanel('firstAttributePanel'),
+      this._emptyAttributePanel('secondAttributePanel')
+    ];
+
+    this.header = ReactWidget.create(
+      <Private.header
+        reloadButton={this._reloadGlueButton}
+        glueCallback={this.glueIdentity}
+      ></Private.header>
+    );
+    this.content = new Widget({ node: this._linkingContent() });
+
+    this._sharedModel.datasetChanged.connect(this.onDatasetsChange, this);
+    this._linkEditorModel.currentDatasetsChanged.connect(
+      this.onDatasetChange,
+      this
+    );
+
+    if (this._linkEditorModel.currentDatasets) {
+      this.onDatasetChange(
+        this._linkEditorModel,
+        this._linkEditorModel.currentDatasets
+      );
+    }
+
+    this._identityAttributes = {
+      first: undefined,
+      second: undefined
+    };
+
+    this.updateDatasets();
+  }
+
+  /**
+   * Update the datasets list.
+   */
+  onDatasetsChange(): void {
+    this.updateDatasets();
+  }
+
+  /**
+   * Reload the panels when the selected datasets changed.
+   *
+   * @param _sender - the link editor model.
+   * @param selection - the selected datasets.
+   */
+  onDatasetChange = (_sender: ILinkEditorModel, selection: IDatasets): void => {
+    // this.updateAttributes();
+    // this.updateAdvancedLink();
+  };
+
+  // /**
+  //  * Update the advanced link panel when the selected advanced link changed.
+  //  *
+  //  * @param selectedLink - the selected advanced link.
+  //  */
+  // onAdvancedLinkChanged = (
+  //   selectedLink: Private.IAdvancedLinkSelected
+  // ): void => {
+  //   this._selectedAdvLink = selectedLink;
+  //   this.updateAdvancedLink();
+  // };
```

Comment:
This block is for the future advanced link?

Reply:
Yes, all these blocks are related to the previous advanced link panel and advanced link summary.
I didn't want to drop all of these and start from nothing when we'll implement the advanced links.
But I guess I can remove them from now and I'll watch in commits history if necessary.

Reply:
No worry, let's keep it here for future reference